src/charm.py

```diff
@@ -419,7 +419,7 @@ def _on_relation_joined(self, event: RelationJoinedEvent) -> None:
         # app relations should be made aware of the new set of hosts
         try:
             self.client_relations.update_app_relation_data()
-            self.shard_relations._update_mongos_hosts()
+            self.config_server._update_mongos_hosts()
```

Comment:
If we are using `_update_mongos_hosts()` outside `ShardingProvider` I think it is better to rename it to `update_mongos_hosts` because "_" prefix is in general for private methods\fields

Reply:
`config_server ` belongs to [sharding Provider now](https://github.com/canonical/mongodb-operator/blob/2b6db034ae7a2f07ce9afc5609da70f5607fd4ed/src/charm.py#L130)